src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -546,13 +551,12 @@ public int MaxResponseDrainSize
 
         protected override void Dispose(bool disposing)
         {
-            if (disposing && !_disposed)
+            if (!_disposed)
```

Comment:
As discussed offline: we need to consider the `disposing` parameter for the Async case. WinHttp will perform an _async_ release of the handle that needs to be _awaited_ (improper reuse of the TPL term) by the callback. Since criticalhandles get released after managed resources, in my pressure tests for WebSockets I've noticed that the AppDomain will go away before the callback gets called, causing a MDA to fire as the delegate (even if marked as static) is not available anymore.